test/numerical/TestLSTM.cpp

```diff
@@ -89,6 +96,17 @@ int main(int argc, char *argv[]) {
           assert(isOMLSTMTheSameAsNaiveImplFor(-1, s, b, i, h, true, true));
           // bidirectional
           assert(isOMLSTMTheSameAsNaiveImplFor(2, s, b, i, h, true, true));
+
+          // Static dimensions (initial_h, initial_c and p are not specified)
+          // forward
+          assert(isOMLSTMTheSameAsNaiveImplFor(
+              1, s, b, i, h, false, false, true, true, true));
+          // reverse
+          assert(isOMLSTMTheSameAsNaiveImplFor(
+              -1, s, b, i, h, false, false, true, true, true));
+          // bidirectional
+          assert(isOMLSTMTheSameAsNaiveImplFor(
+              2, s, b, i, h, false, false, true, true, true));
```

Comment:
It looks like we have more and more repeated calls, it is time to use `for` loops for dynamicS, dynamicB, initial_h, initial_c, and p.